transparency/boot_entry.go

```diff
@@ -0,0 +1,220 @@
+// Copyright (c) The go-boot authors. All Rights Reserved.
+//
+// Use of this source code is governed by the license
+// that can be found in the LICENSE file.
+
+// Package transparency implements an interface to the
+// boot-transparency library functions to ease boot bundle
+// validation.
+package transparency
+
+import (
+	"crypto/sha256"
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+
+	"github.com/usbarmory/boot-transparency/artifact"
+	"github.com/usbarmory/boot-transparency/engine/sigsum"
+	"github.com/usbarmory/boot-transparency/policy"
+	"github.com/usbarmory/boot-transparency/transparency"
+)
+
+// Artifact represents a boot artifact.
+type Artifact struct {
+	// Category must be consistent with the artifact categories
+	// supported by boot-transparency library.
+	Category uint
+
+	// SHA-256 hash of the artifact.
+	Hash string
+}
+
+// BootEntry represent a boot antry as a set of artifacts.
+type BootEntry []Artifact
+
+// Validate the transparency inclusion proof and, the consistency
```

Comment:
`Validate validates ...`, the fact that it takes "as input the pointer" is obvious from the function prototype and doesn't need explaining, same thing for the error response.

I'd suggest a simper rephrase which focuses on the what and leaves the how as a side note:

`Validate applies boot-transparency validation (e.g. inclusion proof, boot policy and claims consistency) for the argument [Config] representing the boot artifacts.`